404.html

```diff
@@ -12,6 +12,62 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+
+
+    <!--This block of code is temporary to account for old links since the site was reorganized.
+    This wont redirect _all_ broken links but catches the main ones.-->
+
+    <!--[if lt IE 9]><script type="text/javascript">var IE_fix=true;</script><![endif]-->
+    <script type="text/javascript">
+        
+        window.onload = function ()
+        {
+            var delay = "2000";
+            setTimeout(GoToURL, delay);
+        }
+        function GoToURL()
+        {
+            if(typeof IE_fix != "undefined") // IE8 and lower fix to pass the http referer
+            {
+                var url = "/docs";
+                var referLink = document.createElement("a");
+                referLink.href = url;
+                document.body.appendChild(referLink);
+                referLink.click();
+            }
+            else { 
+                var pathname = window.location.pathname;
+
+                // redirect how-to/build to build
+                if(pathname.includes("/docs/how-to/build")){
+                    window.location.replace("https://onnxruntime.ai/docs/build");
+                }
+                // how-to was removed - this is a catch all for deleted paths to how-to
+                else if(pathname.includes("/docs/how-to")){
+                    window.location.replace("https://onnxruntime.ai/docs");
+                }
+                else if(pathname.includes("/docs/resources/graph-optimizations")){
+                    window.location.replace("https://onnxruntime.ai/docs/performance/graph-optimizations");
+                }
+                // resource was removed - this is a catch all for deleted paths to resource
+                else if(pathname.includes("/docs/resource")){
+                    window.location.replace("https://onnxruntime.ai/docs/reference");
+                }
+                else if(pathname.includes("/docs/reference/execution-providers")){
+                    window.location.replace("https://onnxruntime.ai/docs/execution-providers/");
+                }
+                else if(pathname.includes("docs/reference/api")){
+                    window.location.replace("https://onnxruntime.ai/docs/api");
+                }
+                // Catch all docs 404 and redirect to docs home
+                else if(pathname.includes("/docs")){
```

Comment:
Do we want this? It might occlude genuine 404s that we need to fix?

Reply:
We can remove this if you want, it would account for any "onnxruntime.ai/docs/X" 404, I added because I know I dont have a redirect for every missing link. Its up to you what you think is best here. I intend to remove this entire script after a few weeks because hopefully everything will be updated and search results reindexing will be updated as well. Let me know what you want me to do!

Reply:
I'd vote for removing it, and explicitly adding paths that we've redirected. There's no harm in leaving the javascript there. We could even add a list of redirects and use some generic code to parse it.

Reply:
removed the catch all added a json file with redirects